python/eos/tasks.py

```diff
@@ -0,0 +1,196 @@
+    The output files will be stored in EOS_BASE_DIRECTORY/POSTERIOR/clusters.
+
+    :param analysis_file: The name of the analysis file that describes the named posterior.
+    :type analysis_file: str
+    :param posterior: The name of the posterior.
+    :type posterior: str
+    :param base_directory: The base directory for the storage of data files. Can also be set via the EOS_BASE_DIRECTORY environment variable.
+    :type base_directory: str, optional
+    :param threshold: The R value threshold. If two sample subsets have an R value larger than this threshold, they will be treated as two distinct clusters. Defaults to 2.0.
+    :type threshold: float > 1.0, optional
+    :param K_g: The number of mixture components per cluster. Default to 1.
+    :type K_g: int >= 1, optional
+    """
+
+    import pathlib
+    output_path = os.path.join(base_directory, posterior, 'clusters')
+    input_paths = [str(p) for p in pathlib.Path(os.path.join(base_directory, posterior)).glob('mcmc-*')]
+    chains    = [eos.data.MarkovChain(path).samples for path in input_paths]
+    means     = [_np.mean(chain, axis=0) for chain in chains]
+    variances = [_np.var(chain,  axis=0) for chain in chains]
+    groups    = pypmc.mix_adapt.r_value.r_group(means, variances, len(chains[0]), critical_r=threshold)
+    eos.info('Found {} groups using an R value threshold of {}'.format(len(groups), threshold))
+    density   = pypmc.mix_adapt.r_value.make_r_gaussmix(chains, K_g=K_g, critical_r=threshold)
+    eos.data.MixtureDensity.create(output_path, density)
+
+
+# Sample PMC
+def sample_pmc(analysis_file, posterior, base_directory='./', step_N=500, steps=10, final_N=5000):
+    """
+    Samples from a named posterior using the Population Monte Carlo (PMC) methods.
+
+    The results of the find-cluster command are expected in EOS_BASE_DIRECTORY/POSTERIOR/clusters.
+    The output file will be stored in EOS_BASE_DIRECTORY/POSTERIOR/pmc.
+
+    :param analysis_file: The name of the analysis file that describes the named posterior.
+    :type analysis_file: str
+    :param posterior: The name of the posterior.
+    :type posterior: str
+    :param base_directory: The base directory for the storage of data files. Can also be set via the EOS_BASE_DIRECTORY environment variable.
+    :type base_directory: str, optional
+    :param step_N: The number of samples to be used in each adaptation step. These samples will be discarded. Defaults to 500.
+    :type step_N: int > 0, optional
+    :param steps: The number of adaptation steps, which ared used to adapt the PMC proposal to the posterior. Defaults to 10.
+    :type steps: int > 0, optional
+    :param final_N: The number of samples to be stored in the output file. Defaults to 5000,
+    :type final_N: int > 0, optional
+    """
+
+    _analysis_file = eos.AnalysisFile(analysis_file)
+    analysis = _analysis_file.analysis(posterior)
+    rng = _np.random.mtrand.RandomState(1701)
```

Comment:
ditto

Reply:
Want this to be reproducible. Let's leave it as is, and discuss how to ensure reproducability, e.g., by setting the seed in the analysis.yaml.